core/net/ip/tcpip.c

```diff
@@ -185,46 +185,32 @@ check_for_tcp_syn(void)
 static void
 packet_input(void)
 {
-#if UIP_CONF_IP_FORWARD
   if(uip_len > 0) {
+
+#if UIP_CONF_IP_FORWARD
     tcpip_is_forwarding = 1;
-    if(uip_fw_forward() == UIP_FW_LOCAL) {
+    if(uip_fw_forward() != UIP_FW_LOCAL) {
       tcpip_is_forwarding = 0;
-      check_for_tcp_syn();
-      uip_input();
-      if(uip_len > 0) {
-#if UIP_CONF_TCP_SPLIT
-        uip_split_output();
-#else /* UIP_CONF_TCP_SPLIT */
-#if NETSTACK_CONF_WITH_IPV6
-        tcpip_ipv6_output();
-#else
-	PRINTF("tcpip packet_input forward output len %d\n", uip_len);
-        tcpip_output();
-#endif
-#endif /* UIP_CONF_TCP_SPLIT */
-      }
+      return;
     }
     tcpip_is_forwarding = 0;
-  }
-#else /* UIP_CONF_IP_FORWARD */
-  if(uip_len > 0) {
+#endif /* UIP_CONF_IP_FORWARD */
+
     check_for_tcp_syn();
     uip_input();
     if(uip_len > 0) {
 #if UIP_CONF_TCP_SPLIT
       uip_split_output();
 #else /* UIP_CONF_TCP_SPLIT */
-#if NETSTACK_CONF_WITH_IPV6
+#if UIP_CONF_IPV6
```

Comment:
`UIP_CONF_IPV6` is deprecated, `NETSTACK_CONF_WITH_IPV6` is to be used as a replacement